bfillings/mafft_v7.py

```diff
@@ -0,0 +1,356 @@
+
+        # Output
+
+        # Output format: clustal format. Default: off (fasta format)
+        '--clustalout': FlagParameter(Prefix='--', Name='clustalout'),
+
+        # Output order: same as input. Default: on
+        '--inputorder': FlagParameter(Prefix='--', Name='inputorder'),
+
+        # Output order: aligned. Default: off (inputorder)
+        '--reorder': FlagParameter(Prefix='--', Name='reorder'),
+
+        # Guide tree is output to the input.tree file. Default: off
+        '--treeout': FlagParameter(Prefix='--', Name='treeout'),
+
+        # Do not report progress. Default: off
+        '--quiet': FlagParameter(Prefix='--', Name='quiet'),
+
+        # Input
+
+        # Assume the sequences are nucleotide. Deafult: auto
+        '--nuc': FlagParameter(Prefix='--', Name='nuc'),
+
+        # Assume the sequences are amino acid. Deafult: auto
+        '--amino': FlagParameter(Prefix='--', Name='amino'),
+
+        # Seed alignments given in alignment_n (fasta format) are aligned with
+        # sequences in input. The alignment within every seed is preserved.
+        '--seed': ValuedParameter(Prefix='--', Name='seed', Delimiter=' '),
+    }
+
+    _parameters = {}
+    _parameters.update(_options)
+    _command = "mafft"
+    _suppress_stderr = False
+
+    def _input_as_seqs(self, data):
+        """Format a list of seq as input.
+
+        Parameters
+        ----------
+        data: list of strings
+            Each string is a sequence to be aligned.
+
+        Returns
+        -------
+        A temp file name that contains the sequences.
+
+        See Also
+        --------
+        burrito.util.CommandLineApplication
+        """
+        lines = []
+        for i, s in enumerate(data):
+            # will number the sequences 1,2,3,etc.
+            lines.append(''.join(['>', str(i+1)]))
+            lines.append(s)
+        return self._input_as_lines(lines)
+
+    def _tree_out_filename(self):
+        if self.Parameters['--treeout'].isOn():
+            tree_filename = self._absolute(str(self._input_filename))+'.tree'
+        else:
+            raise ValueError("No tree output file specified.")
+        return tree_filename
+
+    def getHelp(self):
+        """Method that points to the Mafft documentation."""
+
+        help_str = ("See Mafft documentation at:\n"
+                    "http://align.bmr.kyushu-u.ac.jp/"
+                    "mafft/software/manual/manual.html")
+        return help_str
+
+    def _get_result_paths(self, data):
+        result = {}
+        if self.Parameters['--treeout'].isOn():
+            out_name = self._tree_out_filename()
+            result['Tree'] = ResultPath(Path=out_name, IsWritten=True)
+        return result
+
+
+def align_unaligned_seqs(seqs_fp, moltype=DNA, params=None, accurate=False):
+    """Aligns unaligned sequences
+
+    Parameters
+    ----------
+    seqs_fp : string
+        file path of the input fasta file
+    moltype : callable constructor
```

Comment:
`moltype : {skbio.DNA, skbio.RNA, skbio.Protein}`
This is the way lists of options are listed in numpy documentation.